hikari/impl/voice.py

```diff
@@ -63,16 +63,40 @@ def __init__(self, app: traits.BotAware) -> None:
         self._app = app
         self._connections = {}
         self.connections = types.MappingProxyType(self._connections)
-        self._app.event_manager.subscribe(voice_events.VoiceEvent, self._on_voice_event)
+        self._is_alive = False
 
-    async def disconnect(self) -> None:
+    @property
+    def is_alive(self) -> bool:
+        return self._is_alive
+
+    def _check_if_alive(self) -> None:
+        if not self._is_alive:
+            raise errors.ComponentStateConflictError("Component cannot be used while it's not alive")
+
+    async def _disconnect(self) -> None:
         if self._connections:
             _LOGGER.info("shutting down %s voice connection(s)", len(self._connections))
+            # We rely on the assumption that _on_connection_close will be called here rather than explicitly
+            # emptying self._connections.
             await asyncio.gather(*(c.disconnect() for c in self._connections.values()))
 
+    async def disconnect(self) -> None:
```

Comment:
Why 2 functions? Cant we just keep it to 1?

Reply:
Added an _is_closing field to avoid calls being made to the client while it's closing but _disconnect is still needed to avoid a state conflict error being raised when this is called in close

Reply:
We could also change it to allow close to always be called, even if the component is not running, which would probably be best now that I think about it

Reply:
Something that just popped my mind, we dont actually have a disconnect a specific channels, we only have a full disconnect from all channels. Should we change disconnect to allow passing a channel to disconnect from and moving the closing all connections to `close`?

(This would be done in another pr, since it is a bit out of scope for this one)

Reply:
Right now they could just get the specific connection handler from VoiceClient.connections and then call disconnect on that which should have the same behaviour

Reply:
Ok, after this is merged I will open a pr to do this 👍